src/utilcode/configuration.cpp

```diff
@@ -0,0 +1,148 @@
+        if (names[i] != nullptr && values[i] != nullptr)
+        {
+            size_t bufferLength = wcslen(names[i]) + 1;
+            LPWSTR nameBuffer = new WCHAR[bufferLength];
+            wcscpy_s(nameBuffer, bufferLength, names[i]);
+            knobs[numberOfConfigsSet].knobName = nameBuffer;
+
+            bufferLength = wcslen(values[i]) + 1;
+            LPWSTR valueBuffer = new WCHAR[bufferLength];
+            wcscpy_s(valueBuffer, bufferLength, values[i]);
+            knobs[numberOfConfigsSet].knobValue = valueBuffer;
+
+            numberOfConfigsSet++;
+        }
+    }
+}
+
+static LPCWSTR GetConfigurationValue(LPCWSTR name)
+{
+    _ASSERT(name != nullptr);
+    if (name == nullptr || knobs == nullptr)
+    {
+        return nullptr;
+    }
+
+    for (int i = 0; i < numberOfConfigsSet; ++i)
+    {
+        if (wcscmp(name, knobs[i].knobName) == 0)
+        {
+            return knobs[i].knobValue;
+        }
+    }
+
+    return nullptr;
+}
+
+DWORD Configuration::GetKnobDWORDValue(LPCWSTR name, const CLRConfig::ConfigDWORDInfo& dwordInfo)
+{
+    DWORD legacyValue;
+    bool legacyConfigSet = CLRConfig::GetConfigValue(dwordInfo, false /* useDefaultIfNotSet */, true /* acceptExplicitDefaultFromRegutil */, &legacyValue);
+    if (legacyConfigSet)
+    {
+        return legacyValue;
+    }
+
+    LPCWSTR knobValue = GetConfigurationValue(name);
+    if (knobValue != nullptr)
+    {
+        return wcstoul(knobValue, nullptr, 0);
+    }
+
+    return CLRConfig::GetConfigValue(dwordInfo);
+}
+
+DWORD Configuration::GetKnobDWORDValue(LPCWSTR name, DWORD defaultValue)
+{
+    LPCWSTR knobValue = GetConfigurationValue(name);
+    if (knobValue != nullptr)
+    {
+        return wcstoul(knobValue, nullptr, 0);
+    }
+
+    return defaultValue;
+}
+
+LPCWSTR Configuration::GetKnobStringValue(LPCWSTR name, const CLRConfig::ConfigStringInfo& stringInfo)
+{
+    LPCWSTR value = CLRConfig::GetConfigValue(stringInfo);
+    if (value == nullptr)
+    {
+        value = GetConfigurationValue(name);
+    }
+
+    return value;
+}
+
+LPCWSTR Configuration::GetKnobStringValue(LPCWSTR name)
+{
+    return GetConfigurationValue(name);
+}
+
+bool Configuration::GetKnobBooleanValue(LPCWSTR name, const CLRConfig::ConfigDWORDInfo& dwordInfo)
+{
+    DWORD legacyValue;
+    bool legacyConfigSet = CLRConfig::GetConfigValue(dwordInfo, false /* useDefaultIfNotSet */, true /* acceptExplicitDefaultFromRegutil */, &legacyValue);
+    if (legacyConfigSet)
+    {
+        return (legacyValue == 1);
+    }
+
+    LPCWSTR knobValue = GetConfigurationValue(name);
+    if (knobValue != nullptr)
+    {
+        return (wcscmp(knobValue, W("true")) == 0);
+    }
+
+    return (CLRConfig::GetConfigValue(dwordInfo) == 1);
+}
+
+bool Configuration::GetKnobBooleanValue(LPCWSTR name, bool defaultValue)
+{
+    LPCWSTR knobValue = GetConfigurationValue(name);
+    if (knobValue != nullptr)
+    {
+        return (wcscmp(knobValue, W("true")) == 0);
```

Comment:
I think if we're already expecting an exact string for the property key, it's not any more painful to specify the exact value as well. 

As an added bonus, for hosts that use JSON for these configuration parameters, 'true' and 'false' are the idiomatic way of expressing boolean values.